src/client.ts

```diff
@@ -465,96 +484,175 @@ export class Client {
     }
   }
 
-  async generateConfig(
+  generateConfig(
+    options: GenerateConfigOptions,
+    callback: Callback<GenerateConfigOutput>,
+  ): void;
+  generateConfig(
     options?: GenerateConfigOptions,
   ): Promise<GenerateConfigOutput>;
-  async generateConfig(
+  generateConfig(
     input: GenerateConfigInput,
-    options: GenerateConfigInputOptions,
+    options?: GenerateConfigInputOptions,
   ): Promise<GenerateConfigOutput>;
-  async generateConfig(
-    inputOrOptions?: GenerateConfigInput | GenerateConfigOptions,
+  generateConfig(
+    input: GenerateConfigInput,
     options?: GenerateConfigInputOptions,
-  ): Promise<GenerateConfigOutput> {
-    if (options) {
-      const { strategy, ...httpOptions } = options;
-      const input = inputOrOptions as GenerateConfigInput;
+  ): Promise<GenerateConfigOutput>;
+  generateConfig(
+    input: GenerateConfigInput,
+    options: GenerateConfigInputOptions,
+    callback: Callback<GenerateConfigOutput>,
+  ): void;
+  generateConfig(
+    input: GenerateConfigInput,
+    callback: Callback<GenerateConfigOutput>,
+  ): void;
+  generateConfig(
+    inputOrResetOptions?: GenerateConfigInput | GenerateConfigOptions,
+    optionsOrCallback?:
+      | GenerateConfigInputOptions
+      | Callback<GenerateConfigOutput>,
+    callback?: Callback<GenerateConfigOutput>,
+  ): Promise<GenerateConfigOutput> | void {
+    const { cb, opts } = parseFunctionOverloads(optionsOrCallback, callback);
+
+    if (!inputOrResetOptions || 'reset' in inputOrResetOptions) {
+      return respond(() => {
+        const { reset, ...httpOptions } = inputOrResetOptions ?? {};
+        const GET_GENERATE_CONFIG_ID = 'get-generate-config';
+
+        if (reset) {
+          return this.#fetcher<ApiTypes.GenerateConfigOutput>({
+            ...httpOptions,
+            method: 'DELETE',
+            url: '/v1/generate/config',
+            cache: {
+              update: {
+                [GET_GENERATE_CONFIG_ID]: 'delete',
+              },
+            },
+          });
+        } else {
+          return this.#fetcher<ApiTypes.GenerateConfigOutput>({
+            ...httpOptions,
+            method: 'GET',
+            url: '/v1/generate/config',
+            id: GET_GENERATE_CONFIG_ID,
+            cache: {
+              update: {
+                [GET_GENERATE_CONFIG_ID]: 'delete',
+              },
+            },
+          });
+        }
+      }, cb);
+    }
+
+    return respond(() => {
+      const { strategy, ...httpOptions } = opts ?? {};
       return this.#fetcher<
         ApiTypes.GenerateConfigOutput,
         ApiTypes.GenerateConfigInput
       >({
         ...httpOptions,
         method: strategy === 'merge' ? 'PATCH' : 'PUT',
         url: '/v1/generate/config',
-        data: input,
+        data: inputOrResetOptions as GenerateConfigInput,
```

Comment:
We should narrow the input type via the `parseFunctionOverloads` utility as well.